data/account/participationRegistry.go

```diff
@@ -797,6 +853,17 @@ func scanRecords(rows *sql.Rows) ([]ParticipationRecord, error) {
 			}
 		}
 
+		if len(rawStateProof) > 0 {
+			stateProof := merklesignature.Signer{}
+			err = protocol.Decode(rawStateProof, &stateProof.SignerContext)
+			if err != nil {
+				return nil, fmt.Errorf("unable to decode stateproof: %w", err)
+			}
+			var stateProofVerifer StateProofVerifier
+			copy(stateProofVerifer[:], stateProof.GetVerifier()[:])
+			record.StateProof = &stateProofVerifer
```

Comment:
It is weird, as the `Keysets` table do not directly reflect the `ParticipationRecord` type.
The thing is, the `ParticipationRecord` should only contain the verifier, as it is the only part of the StateProof that's relevant to it.
However, when actually using the StateProof keys to sign, we require the rest of the data.

The way it is currently stored is such: the `stateProof BLOB` within the `Keysets` table is basically encoded `merklesignature.SignerContext`, which is the State Proof data without the keys themselves, as the keys are quite large and we prefer to store them separately for performance and security reasons.
The `FalconSigner` keys are stored within `StateProofKeys` table.

Since the `ParticipationRecord` does not reflect how data is actually stored in the database, we have to build it manually and extract only what's necessary - in case of stateproof the verifier, which is the root of the merkle tree built upon the keys.

Some database migration is inevitable either way in my opinion, since the voting keys should not be stored the way that they are, so we could use that migration to also update the way we handle the state proof data, but for now I'm not sure of a better way to do that.

What do you think?

Reply:
Sounds good, I wanted to make sure it wasn't weird because there was some mistake!

A comment with a summary of this explanation might be helpful for the next person working on this code.